src/widget.js

```diff
@@ -86,6 +86,11 @@ export default class Widget extends Plugin {
 		const viewDocument = editor.editing.view;
 		let element = domEventData.target;
 
+		// Widget is not selectable when editor is read-only.
```

Comment:
> Widget might be salectable but only on the model side (like entire editor in read-only mode). But I think this selection should be not visible like caret is not visible when editor is read-only. WDYT?

Pity that it can't be a CSS-only solution... ;/ AFAIR, unfortunately, due to nested editables you're not able to write a CSS which will disable focus and hover styles for the right widgets.

The case is this:

```
disabledEditable >
   enabledEditable >
      widget
```

If you style the `:hover` to not work in  disabledEditable then it's going to break the widget styles in this case (the widget should be selectable because its closest editable is enabled.